src/realm/CMakeLists.txt

```diff
@@ -213,10 +213,24 @@ elseif(ANDROID)
     list(APPEND REALM_EXTRA_LIBS atomic)
 endif()
 
-# The placeholder file is needed in order to make builds work on Xcode
-# The issue is described in the last sentence here:
-# https://cmake.org/cmake/help/v3.7/command/add_library.html#object-libraries
-add_library(realm STATIC $<TARGET_OBJECTS:realm-objects> placeholder.cpp)
+set(REALM_XCODE_DEPENDENCY "")
+if(CMAKE_GENERATOR STREQUAL Xcode)
+    # The placeholder file is needed in order to make builds work on Xcode
+    # The issue is described in the last sentence here:
+    # https://cmake.org/cmake/help/v3.7/command/add_library.html#object-libraries
+    # Another problem is that Xcode does not see changes in the realm-objects
+    # target, to solve this we trigger rebuiliding the library by making
+    # placeholder.cpp depend on all the sources rather than realm-objects.
+    set(REALM_XCODE_DEPENDENCY "placeholder.cpp")
+    add_custom_command(
+        OUTPUT ${REALM_XCODE_DEPENDENCY}
+        DEPENDS ${REALM_SOURCES} ${REALM_INSTALL_ALL_HEADERS}
+        COMMAND echo \"//This file is generated by cmake for Xcode builds\" > ${REALM_XCODE_DEPENDENCY}
+        COMMAND echo \"extern int xcode_unused\;\" >> ${REALM_XCODE_DEPENDENCY}
```

Comment:
Good observation, I changed it to only output one line always overwriting so this shouldn't happen anymore.